gap/lqueue.gi

```diff
@@ -1,174 +1,224 @@
-#############################################################################
-##
-#W  lqueue.gd                   GAP library                  Reimer Behrends
-##
-##
-#Y  Copyright (C) 2013 The GAP Group
-##
-##  This file implements a deque based on a circular buffer. It can be used
-##  to implement FIFO queues as well as stacks.
-##
+# lqueue.gi
+
+# This file implements double ended queues (deques) using a circular buffer
+# stored in a GAP plain list.
+#
+# The four positions in a queue Q have the following purpose
+#
+# Q[1] - head, the index in Q[4] of the first element in the queue
+# Q[2] - tail, the index in Q[4] of the last element in the queue
+# Q[3] - capacity, the allocated capacity in the queue
+# Q[4] - GAP plain list with storage for capacity many entries
+#
+# Global variables QHEAD, QTAIL, QCAPACITY, and QDATA are bound to reflect
+# the above.
+#
+# When a push fills the queue, its capacity is doubled using PlistQueueExpand.
+# A new empty plist is allocated and all current entries of the queue are copied
+# into the new plist with the head entry at index 1.
+#
+# The queue is empty if and only if head = tail and the entry that head and tail
+# point to in the storage list is unbound.
 
 InstallGlobalFunction(PlistQueue,
 function(arg)
-  local capacity, filter, result, i, t;
+    local capacity, factor, result;
 
-  filter := IsPlistQueueRep;
+    if Length(arg) >= 3 then
+        ErrorNoReturn("usage: PlistQueue( [ <capacity>, [ <factor> ] ])");
```

Comment:
This error message is not tested right now (of course we can add a test for it after this PR, so it's not urgent, but I mention it for completeness)